packages/core/src/http-client/csrf-token-header.ts

```diff
@@ -58,7 +62,10 @@ function makeCsrfRequest<T extends HttpRequestConfig>(
   return executeHttpRequest(destination, appendSlash(axiosConfig))
     .then(response => response.headers)
     .catch(e1 => {
-      logger.error(
+      if (hasCsrfToken(e1)) {
```

Comment:
Minor, this was here before, but consider using `error` or `error1` as a variable name.
Also, as we do the same thing twice, execute + return headers + check whether there is csrf token this should be a separate function.

Reply:
Will use `error1`.

I know we can make a function for reusing with the parameters:
- a function to control the request url (with/without trailing slash)
- a warning message (first attempt/second attempt)
- a flag to control whether the next attempt is needed

However, as a work around for the known issue of the axios, I would make it more readable instead of more abstract.
Also, when debugging, I found the current structure is easier, because you know it's first attempt or second within a second.